File: app/src/main/java/com/itsp/android/innovacion2016/operaciones/FunGen.java
package com.itsp.android.innovacion2016.operaciones;


import java.util.Date;

public class FunGen {

    public static double ConvertDoubleStr(String cad)
    {
        cad = cad.trim();
        if (cad.isEmpty())
        {
            return 0.0;
        }
        else
        {
            try
            {
                return Double.parseDouble(cad);
            }
            catch (Exception e)
            {
                return 0.0;
            }
        }
    }

    public static int ConvertIntStr(String cad)
    {
        cad = cad.trim();
        if (cad.equals(null) || cad.trim().isEmpty())
        {
            return 0;
        }
        else
        {
            try
            {
                return Integer.parseInt(cad);
            }
            catch (Exception e)
            {
                return 0;
            }
        }
    }

    public static float ConvertFloatStr(String cad)
    {
        cad = cad.trim();
        if (cad.trim().isEmpty())
        {
            return 0.0f;
        }
        else
        {
            try
            {
                return Float.parseFloat(cad);
            }
            catch (Exception e)
            {
                return 0.0f;
            }
        }
    }

    public static int CadeToInt(String cad)
    {
        cad = cad.trim();
        if (cad.equals(null) || cad.trim().isEmpty())
        {
            return 0;
        }
        else
        {
            try
            {
                return Integer.parseInt(cad);
            }
            catch (Exception e)
            {
                return 0;
            }
        }
    }

}

File: app/src/main/java/com/gcr/android/inventorytest/operaciones/FunGen.kt
package com.gcr.android.inventorytest.operaciones


object FunGen {
    @JvmStatic
    fun ConvertDoubleStr(cad: String): Double {
        var cad = cad
        cad = cad.trim { it <= ' ' }
        if (cad.isEmpty()) {
            return 0.0
        } else {
            try {
                return cad.toDouble()
            } catch (e: Exception) {
                return 0.0
            }
        }
    }

    fun ConvertIntStr(cad: String): Int {
        var cad = cad
        cad = cad.trim { it <= ' ' }
        if (cad == null || cad.trim { it <= ' ' }.isEmpty()) {
            return 0
        } else {
            try {
                return cad.toInt()
            } catch (e: Exception) {
                return 0
            }
        }
    }

    fun ConvertFloatStr(cad: String): Float {
        var cad = cad
        cad = cad.trim { it <= ' ' }
        if (cad.trim { it <= ' ' }.isEmpty()) {
            return 0.0f
        } else {
            try {
                return cad.toFloat()
            } catch (e: Exception) {
                return 0.0f
            }
        }
    }

    fun CadeToInt(cad: String): Int {
        var cad = cad
        cad = cad.trim { it <= ' ' }
        if (cad == null || cad.trim { it <= ' ' }.isEmpty()) {
            return 0
        } else {
            try {
                return cad.toInt()
            } catch (e: Exception) {
                return 0
            }
        }
    }
}